Source/ui_qt/QBootablesView.cpp

```diff
@@ -246,7 +267,7 @@ void QBootablesView::on_awsS3Button_clicked()
 		if(new_entry)
 		{
 			AsyncUpdateStatus("Refreshing Model.");
-			resetModel();
```

Comment:
not sure entirely whats happening here, it seems CC segfault if its gets another future thats registered inside a future?

Reply:
I guess this is because the new future is being registered while we are iterating over the vector here: https://github.com/jpd002/Play-/blob/master/Source/FutureContinuationManager.h#L20. 

If registering within a callback would make code simpler, I wouldn't be against changing the ContinuationManager to handle this (like, we could collect a list of callback inside the `remove_if` iteration instead of calling the callbacks directly).

Reply:
> If registering within a callback would make code simpler

not necessarily, atleast in this very case, we already have the async signal function, so it works fine by just using it, but its something to note incase it becomes a problem in the future.